pipeline_tools/shared/submission/create_analysis_protocol.py

```diff
@@ -0,0 +1,171 @@
+    schema_type = SCHEMAS["ANALYSIS_PROTOCOL"]["schema_type"]
+    schema_version = SCHEMAS["ANALYSIS_PROTOCOL"]["schema_version"]
+    type = {
+        "text": "analysis_protocol"
+    }
+
+    def __init__(
+        self,
+        method,
+        input_uuid,
+        pipeline_type,
+        pipeline_version,
+            workspace_version):
+
+        self.computational_method = method
+        self.input_uuid = input_uuid
+        self.pipeline_type = pipeline_type
+        self.protocol_core = {
+            "protocol_id": pipeline_version
+        }
+        self.work_version = workspace_version
+
+    def __analysis_protocol__(self):
+        return {
+            "computational_method": self.computational_method,
+            "describedBy": self.describedBy,
+            "protocol_core": self.protocol_core,
+            "provenance": self.__get_provenance(),
+            "schema_type": self.schema_type,
+            "type": self.type
+        }
+
+    def get_json(self):
+        return self.__analysis_protocol__()
+
+    # Return json to hash for entity_id
+    def __get_hash_input(self):
+        return {
+            "computational_method": self.computational_method,
+            "describedBy": self.describedBy,
+            "protocol_core": self.protocol_core,
+            "schema_type": self.schema_type,
+            "type": self.type
+        }
+
+    # Generate provenance
+    def __get_provenance(self):
+        string_to_hash = json.dumps(self.__get_hash_input())
+        entity_id = format_map.get_uuid5(string_to_hash)
+
+        return {
+            "document_id": entity_id,
+            "submission_date": self.workspace_version,
+            "update_date": self.workspace_version
+        }
+
+    @property
+    def entity_id(self):
+        provenance = self.__get_provenance()
+        entity_id = provenance["document_id"]
+        return entity_id
+
+    @property
+    def workspace_version(self):
+        return self.work_version
+
+
+# Entry point for unit tests
+def test_build_analysis_protocol(
+    method,
+    input_uuid,
+    pipeline_type,
+    pipeline_version,
+        workspace_version):
+
+    test_analysis_protocol = AnalysisProtocol(
+        method,
+        input_uuid,
+        pipeline_type,
+        pipeline_version,
+        workspace_version
+    )
+    return test_analysis_protocol.get_json()
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--pipeline_type", required=True, help="Type of pipeline(SS2 or Optimus)")
+    parser.add_argument("--input_uuid", required=True, help="Input file UUID from the HCA Data Browser")
+    parser.add_argument("--workspace_version", required=True, help="Workspace version value i.e. timestamp for workspace")
+    parser.add_argument(
+        "--method",
+        required=True,
+        help="Supposed to be method store url, for now can be url for wdl in skylab, or the name of"
```

Comment:
I believe the dockstore URL is the correct method store URL

Reply:
I think we might also be able to generate the Dockstore URL with the pipeline version